lib/rules/no-mocha-arrows.js

```diff
@@ -50,7 +50,41 @@ module.exports = function (context) {
                 fnArg = node.arguments.slice(-1)[0];
                 if (fnArg && fnArg.type === 'ArrowFunctionExpression') {
                     if (isLikelyMochaGlobal(context.getScope(), name)) {
-                        context.report(node, 'Do not pass arrow functions to ' + name + '()');
+                        context.report({
+                            node: node,
+                            message: 'Do not pass arrow functions to ' + name + '()',
+                            fix: function (fixer) {
+                                var sourceCode = context.getSourceCode(),
+                                    paramsLeftParen = sourceCode.getFirstToken(fnArg),
+                                    paramsRightParen = sourceCode.getTokenBefore(sourceCode.getTokenBefore(fnArg.body)),
+                                    paramsFullText =
+                                        sourceCode.text.slice(paramsLeftParen.range[0], paramsRightParen.range[1]),
+                                    functionKeyword = 'function ',
+                                    bodyText;
+
+                                if (fnArg.async) {
+                                    // When 'async' specified, take care about the keyword.
+                                    functionKeyword = 'async ' + functionKeyword;
+                                    // Strip 'async (...)' to '(...)'
+                                    paramsFullText = paramsFullText.slice(6);
```

Comment:
Both `async () => {}` and `async() => {}` are actually valid. This would create broken code if you find the latter form. (add a test for that obviously :) )